### DISCUSSION TITLE
2021-12-13 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#882] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Frontend priorities](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792220)
- [Backend priorities](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792230)
- [Design priorities](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792234)
- [Product priorities](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792238)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792242)
- [Sean's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1794099)
- [Mukesh's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1796179)
- [Ghislaine's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1797628)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1797735)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1805962)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1805396)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1822772)

## Weekly check-in
@dmos62 @ghislaineguerin @mathemancer @pavish @seancolsen @silentninja please respond to the weekly check-in as a top-level response and edit the discussion topics above to add your check in as a topic. Others are also welcome to respond.

- How did the last week go? What went well, what could have gone better?
- What are you planning to work on over the next week? Do you anticipate any blockers?
- Anything else you'd like to add?

**Comment by kgodey** [comment]
## Frontend priorities

1. Review and merge the large number of PRs waiting for review. We didn't make any progress on this last week because of other priorities, but hopefully we can get it done this week, especially since @pavish is back.
    1. #836 
    2. #844 
    3. #852 
    4. #855 
    5. #857 
    6. #860 
    7. #861 
    8. #868 
    9. #870 
2. Frontend changes to unblock #862 when API discussions there are complete.
3. Continue to make progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone.

@pavish @seancolsen Any concerns?

**Reply by seancolsen** [reply]
Sounds good. I've assigned myself #862. Will begin working on that once someone notifies me that the API changes in there are approved/finalized.

**Reply by seancolsen** [reply]
Regarding my ongoing [TS strict](https://github.com/centerofci/mathesar/pull/826) PR

- If we get to a point where we don't have any major front-end PRs open and we still have some time in the week, I'd also like to try and get this finished/merged.

- I've been chipping away at the easy errors. I've made a lot of progress, but there's still a tad more work for me to do just on the easy error. After that I'll need some help from Pavish to complete the PR it because there are some harder errors that I'm not sure how best to resolve without more knowledge of the intent behind the code. I was even thinking it might be a good opportunity for us to do some pair programming, if you'd be up for that @pavish. I'd be open to driving or navigating. And no pressure.

- I'm still not worried too much about merge conflicts in this PR. I know I've done some whining about conflicts before, but I see this PR a bit differently. Most of the changes in it are straightforward, localized, and independent -- so in resolving a conflict I'd err very strongly towards accepting the changes within the _other_ PR. I'll continue to keep it up do date with master this week as we hopefully see master move forward a good bit.

**Reply by pavish** [reply]
Sounds good to me.

@seancolsen Regarding the TS strict PR, sure, I'd be up for some pair programming if you feel that it'd make it easier for you to complete the PR.

**Comment by kgodey** [comment]
## Backend priorities

1. Get the following major PRs merged into `master`. Please be proactive about communicating blockers for yourself and others.
    1. #628
    2. #837 
    3. #862 
    4. #865
2. #883 - someone needs to take this, it's currently unassigned.
3. Continue to make progress on other issues in the [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11) milestone.

@dmos62 @mathemancer @silentninja Any concerns?

**Reply by seancolsen** [reply]
Similar to #883, it would be great if someone could take #859 this week because that's blocking some front end work on #775 within the Working with Tables milestone that we're trying to close out soon.

**Reply by silentninja** [reply]
I am interested in taking up #883. I will be following the implementation details we discussed in issue #623.

**Reply by mathemancer** [reply]
I think the priorities make sense.

**Reply by dmos62** [reply]
I think the priorities make sense.

**Comment by kgodey** [comment]
## Design priorities

1. Get the two active specs merged. @ghislaineguerin, please be proactive about resolving any blockers.
    1. https://github.com/centerofci/mathesar-wiki/pull/13.
    2. https://github.com/centerofci/mathesar-wiki/pull/12
2. Make progress on design for:
    1.  https://github.com/centerofci/mathesar/issues/455
    2. https://github.com/centerofci/mathesar/issues/466

@ghislaineguerin @ppii775 Any concerns?

**Reply by ppii775** [reply]
@kgodey @ghislaineguerin Sounds good to me! I'll get a few discussion items ready this week regarding #466 for Ghislaine to review and approve.

**Reply by ghislaineguerin** [reply]
@kgodey got it. I'll focus on closing the active specs.

**Comment by kgodey** [comment]
## Product priorities

I was tied up with working on administrative, design, and backend tasks last week, so my priorities remain unchanged last week:

1. Define Mathesar Types better: https://github.com/centerofci/mathesar-wiki/issues/14
2. Define data sources in Views better and update other Views tickets: #806
3. Create backend and frontend issues for merged design specs once merged.

**Reply by seancolsen** [reply]
I'd be happy to create the front-end tickets for merged design specs if that would be helpful to you, @kgodey

**Reply by kgodey** [reply]
That would be really helpful @seancolsen, thanks!

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2021-12-13 to 2021-12-17 | Sean |
| 2021-12-20 to 2021-12-23 | Dominykas |
| 2021-12-24 to 2022-01-02 | *CCI closed; team is unavailable.* |
| 2022-01-03 to 2022-01-07 | Ghislaine |
| 2022-01-10 to 2022-01-14 | Brent |
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |
| 2022-01-31 to 2022-02-04 | Mukesh |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Reply by pavish** [reply]
I can take up comms during _2021-12-24 to 2022-01-02_ as I already took my week off and I don't celebrate Christmas.

**Reply by kgodey** [reply]
@pavish Thanks! I'll put you on the schedule in next week's update.

**Comment by seancolsen** [comment]
## Sean's check in

How did the last week go? What went well, what could have gone better?

- Going well.
- So far it hasn't been a problem to be so far out ahead of PR review. Interested to see how it goes this week once all the reviews start coming in and I need to make changes to PRs upon which others are based.
- I'm liking the new "Priorities" threads in our weekly planning threads even more as we continue to use them.

What are you planning to work on over the next week? Do you anticipate any blockers?

- Issues in the [Working with Tables](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22%5B07%5D+Working+with+Tables%22+-label%3A%22type%3A+meta%22) milestone.

Anything else you'd like to add?

- Nope.

**Comment by silentninja** [comment]
> How did the last week go? What went well, what could have gone better?

It was okay, a few issues took longer than expected. Comms schedule went well, was able to get used to the process of triaging issues and PR.  I was also able to review the design PR's and get some knowledge of the frontend code and make some progress in writing documentation about how the code is organised during the weekends.

Could have been better if we were able to finish off the issues and PR as planned but happy that we will come to a conclusion this week on the PR's.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Aside from the priorities listed above, I am planning to work on 
- #428 and #433
- #859 and #883 If no one else is interested 
- #430 
- #839 
If #865  gets delayed it might affect #428, but the review changes are very few, so I am hoping it won't take that long to get merged. 

> Anything else you'd like to add?

None at the moment

**Reply by dmos62** [reply]
How do you feel about your workload? I could take #859 and #883 off your hands, if you like.

**Reply by silentninja** [reply]
@dmos62 I put together a meta issue #889 that lists all the requirements for error handling. #883 and #623 can be worked on in parallel. I will be working on #623 first, please feel free to unassign me on #883 and #859 and assign it yourself if you have time and are interested to work on it.

**Reply by dmos62** [reply]
@silentninja Thanks for the rundown. I've something I'd like to figure out in filtering at the moment. This will be next on my todo list unless something changes.

**Comment by ghislaineguerin** [comment]
> How did the last week go? What went well, what could have gone better?

Last week was not very productive as I had a lot going on and was hard to focus. I would have liked to close the PRs. I replied and made changes, but should have listed all that was left and completed all at once, rather than partially. 

> What are you planning to work on over the next week? Do you anticipate any blockers?

I plan on starting a PR for 'Listing of Views' and starting design work for another issue under the same milestone.

> Anything else you'd like to add?

Just that I think I can move faster now that I'm not traveling and in my usual schedule.

**Comment by pavish** [comment]
## Pavish's checkin

> How did the last week go? What went well, what could have gone better?

N/A. I was off for the entire week.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Getting all the open frontend PRs merged, and making progress on the [Working with Tables](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22%5B07%5D+Working+with+Tables%22+-label%3A%22type%3A+meta%22) milestone.

> Anything else you'd like to add?

None, at the moment.

**Comment by mathemancer** [comment]
## Brent's Check in

> How did the last week go? What went well, what could have gone better?

Things went well overall.  I think we're making good progress on getting some of the back end PRs merged, though it's requiring lots of somewhat slow discussions.  I think we are running into some issues with dependent PRs, though.  We'll need to focus on that.

> * What are you planning to work on over the next week? Do you anticipate any blockers?

I want to get the grouping PR merged, and then the filtering PRs merged.  I also have some other coding to do on ranged grouping before it's totally complete as per the spec.  I'm planning to pick up some smaller tickets (e.g., the boolean type) when I have time between longer-term issues.

> * Anything else you'd like to add?

Not at the moment.

**Reply by dmos62** [reply]
> I also have some other coding to do on ranged grouping before it's totally complete as per the spec.

Are you planning to include these changes into current PR, or will that be a new PR?

**Reply by mathemancer** [reply]
It will be a new PR.  I want to get the current one merged ASAP so it unblocks the dependent PRs.  The further changes are much simpler, and create less disruption (it's just a couple more grouping modes), so I think they can wait till after all the other stuff is merged in.

**Comment by dmos62** [comment]
> How did the last week go? What went well, what could have gone better?

Went good. Got some concrete work done on the filtering front, which feels nice. I'm happy with the changes too.

I've something to note. After finishing the number filtering task, and taking up string matching/filtering task, I was blindsided by requirements I had not anticipated (namely, settings; namely, a case-sensitivity setting). Required refactor was minimal, but if I had taken into account more complex filtering cases from the beginning, I would have been better prepared.

So, this is a note for myself to not overly-focus on the current task only. Also, this is a general reflection that there's significant merits to starting with the most complex cases, instead of the simplest cases, even though the "learning curve" is more intimidating. @kgodey I think you might find above train of thought interesting (since you do a lot of planning).
    
> What are you planning to work on over the next week? Do you anticipate any blockers?

Continue working through filtering tasks. Some backend reviews, though I don't anticipate much of that being needed. Adding finishing touches to the first fitlering PR, once range grouping PR is merged/finalized.

> Anything else you'd like to add?

Nope.

**Comment by kgodey** [comment]
> How did the last week go? What went well, what could have gone better?

It went pretty well. I focused mainly on the board presentation. Most of the work there was trying to figure out how to talk about Mathesar to a non-technical audience that's not at all familiar with the project. It took me some time to work that out (I'm usually so focused on the day to day), but I think is going to be useful as we get closer to releasing the product.

I did get behind on code reviews, design reviews, and product work. There's not much of the week left so I don't think I'll be able to catch up on product work, but I can catch up on the other things.

> What are you planning to work on over the next week? Do you anticipate any blockers?

Code reviews, design reviews, and [product priorities](https://github.com/centerofci/mathesar/discussions/882#discussioncomment-1792238) if time permits.
